tests/Integration/Behaviour/Features/Context/CategoryFeatureContext.php

```diff
@@ -0,0 +1,209 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace Tests\Integration\Behaviour\Features\Context;
+
+use Category;
+use Configuration;
+use Group;
+use RuntimeException;
+use Tests\Integration\Behaviour\Features\Context\Util\PrimitiveUtils;
+
+class CategoryFeatureContext extends AbstractPrestaShopFeatureContext
+{
+    /**
+     * @var int
+     */
+    private $defaultLanguageId;
+
+    public function __construct()
+    {
+        $this->defaultLanguageId = (int) Configuration::get('PS_LANG_DEFAULT');
+    }
+
+    /**
+     * @Given /^I specify "(.+)" "(.+)" for new category "(.+)"$/
+     */
+    public function specifyProperty($property, $value, $reference)
+    {
+        $propertiesMap = [
+            'name' => 'name',
+            'displayed' => 'active',
+            'parent category' => 'parent_category_id',
+            'description' => 'description',
+            'meta title' => 'meta_title',
+            'meta description' => 'meta_description',
+            'friendly url' => 'link_rewrite',
+        ];
+
+        if (!isset($propertiesMap[$property])) {
+            throw new RuntimeException(sprintf('Property "%s" is not defined', $property));
+        }
+
+        $propertiesKey = sprintf('%s_properties', $reference);
+        $properties = SharedStorage::getStorage()->getWithDefault($propertiesKey, []);
+
+        if ('parent_category_id' === $propertiesMap[$property]) {
+            $categoriesMap = [
+                'Home Accessories' => 8,
+            ];
+
+            $value = $categoriesMap[$value];
+        }
+
+        $properties[$propertiesMap[$property]] = $value;
+
+        SharedStorage::getStorage()->set($propertiesKey, $properties);
+    }
+
+    /**
+     * @Given /^I specify displayed to be "(enabled|disabled)" for new category "(.+)"$/
+     */
+    public function specifyDisplayedProperty($value, $reference)
+    {
+        $propertiesKey = sprintf('%s_properties', $reference);
+        $properties = SharedStorage::getStorage()->getWithDefault($propertiesKey, []);
+
+        $properties['is_enabled'] = 'enabled' === $value;
+
+        SharedStorage::getStorage()->set($propertiesKey, $properties);
+    }
+
+    /**
+     * @Given /^I specify group access for "(.+)" for new category "(.+)"$/
+     */
+    public function specifyGroupAccess($value, $reference)
+    {
+        $propertiesKey = sprintf('%s_properties', $reference);
+        $properties = SharedStorage::getStorage()->getWithDefault($propertiesKey, []);
+
+        $groupsMap = [
+            'Visitor' => 1,
+            'Guest' => 2,
+            'Customer' => 3,
```

Comment:
@matks is it ok to use hardcoded ids? They are part of demo data.

Reply:
Not a good idea. Functional tests must be able to be executed in prod environement. It's always better to setup and cleanup generated data.

Reply:
Currently, the database reset function being used loads some data so we can use it IMO. But if one day we clean up the reset database processing and remove the fixtures loading (instead we only load an empty database schema), there we will have to load it by ourselves.

Can you add a comment to write this down, in order to explain this to other developers who would work on this test ?